pkg/ingester/index.go

```diff
@@ -77,16 +77,17 @@ func (shard *indexShard) lookup(matchers []*labels.Matcher) []model.Fingerprint
 		if !ok {
 			return nil
 		}
-		var toIntersect []model.Fingerprint
+		var toIntersect model.Fingerprints
 		if matcher.Type == labels.MatchEqual {
 			fps := values[model.LabelValue(matcher.Value)]
-			toIntersect = merge(toIntersect, fps)
+			toIntersect = append(toIntersect, fps...)
```

Comment:
Small thing: This step would be more explicit without the append, just skimming through I expected a Sort to follow an append eventually.

Reply:
Yeah, I thought there must be some reason why it was calling `merge()` before, which is called out as a "special case" above, so I left it with the allocate/copy. But now I look more closely I can't see it.

Reply:
There is a reason: the `fps` must be accessed under lock, so we must take a copy before leaving the function. 

`intersect()` won’t copy if there is nothing to intersect.

Reply:
Well, that's not very obvious, but makes sense. Perhaps a comment about that, otherwise this LGTM